handler,webhook:use variable to set replicas

alonSadan opened the pull request:
<!-- Thanks for sending a pull request!
Before you click the 'Create pull request' make sure that:
- This PR introduces a single feature of fix, just one
- This PR does not leave the main branch broken
- Every commit in this PR has a commit message explaining what do you change,
  why and what is the outcome
- If your change introduces a complex concept or a change to user interaction
  with the project or the application, make sure to document it
If you don't comply with these rules, you waste your energy, time of reviewers
and cause suffering of future generations.
-->

**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What this PR does / why we need it**:
In order to set the number of webhook replicas dynamically, the operator
yaml should accept a variable to set number of replicas. An example use-case
for that would be to set the number of webhook replicas to 1 when running
on a Single-Node-Openshift cluster, and 2 otherwise.


Signed-off-by: alonsadan <asadan@redhat.com>

**Special notes for your reviewer**:
This PR leaves the the number of replicas as 2, and only defines the variable with a fixed value. In a follow-up PR this could be modified to be set dynamically.
This PR si originated in @phoracek  request [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/973#discussion_r693719016)
**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
3. If no release note is required, just write "NONE".
-->

```release-note
NONE
```

rhrazdil commented:
/ok-to-test

alonSadan commented:
/retest

rhrazdil commented on controllers/nmstate_controller.go:
```diff
@@ -189,6 +189,7 @@ func (r *NMStateReconciler) applyHandler(instance *nmstatev1beta1.NMState) error
 	data.Data["CAOverlapInterval"] = ""
 	data.Data["CertRotateInterval"] = ""
 	data.Data["CertOverlapInterval"] = ""
+	data.Data["WebhookReplicas"] = int32(2)
```
Could you please move this line next to other Webhook variables? Either before `WebhookNodeSelector` or after `WebhookAffinity`.
Also, it would be nice if you could define a WEBHOOK_REPLICAS constant

alonSadan replied:
[Done.](https://github.com/nmstate/kubernetes-nmstate/compare/bbbb0a1989a87e288fc800d2f875f00d072f3c45..be645d50e4d2f2b3c16050ee4f82a9e5f33e73b2)

alonSadan commented:
Thanks you @rhrazdil . [Addressed your comments.](https://github.com/nmstate/kubernetes-nmstate/compare/bbbb0a1989a87e288fc800d2f875f00d072f3c45..be645d50e4d2f2b3c16050ee4f82a9e5f33e73b2)

RamLavi reviewed (approved):
/lgtm

phoracek commented:
/approve

phoracek commented:
/hold

rhrazdil commented:
Thanks!
/lgtm

/unhold

alonSadan commented:
Thanks @phoracek @rhrazdil

phoracek commented:
We will need to release knmstate in order to propagate this to CNAO. Two patches were added since the last tag: this and Alona's squash of controllers. Our release lane is currently broken due to missing credentials, but I think I can push it manually. @rhrazdil do you think we are in a good spot to tag v0.53.0?

phoracek commented:
Maybe we could try to get the VLAN PR in to make it more interesting :eyes:
